### DISCUSSION TITLE
how to  include the auto generated traceid  in the http response head?

### DISCUSSION THREAD

**Discussion by szhengli** [apache/skywalking#9699] [General]
I would include the unique  traceid,  auto-generated  by java agent,  in the http response head.  In this way,  SREs   may accurately trace a specific  http request  processing, according to the  traceID included in the response, as a customized header,  say   x-traceID, in browser.   may anyone advises how ?   much appreciated.

**Comment by wu-sheng** [comment]
That requires code-level changes, as we don't plan to manipulate the response. Because response could be very complex, such as binary streaming(media), pic, and audio. 

You need to change codes and use this API to get the current trace ID and inject it into any place you like
https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/application-toolkit-trace/

**Reply by szhengli** [reply]
yes, did it. LOL. thanks

![image](https://user-images.githubusercontent.com/16127498/192963582-182906c6-7f36-4146-b095-6ee8e908fbb5.png)
